Weather/Weather/Controllers/PageViewController.swift

```diff
@@ -0,0 +1,88 @@
+
+//
+//  PageViewController.swift
+//  Weather
+//
+//  Created by Sergii Lisnyi on 5/16/18.
+//  Copyright © 2018 Sergii Lisnyi. All rights reserved.
+//
+import UIKit
+import CoreLocation
+
+class PageViewController: UIPageViewController {
+    
+    var modelCities = ModelCities()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadDataWeather()
+        dataSource = self
+    }
+    
+    func isEnableLocation() -> Bool {
+        guard CLLocationManager.locationServicesEnabled() else { return false }
+        return [.authorizedAlways, .authorizedWhenInUse].contains(CLLocationManager.authorizationStatus())
+    }
+    
+    func displayController(index: Int) -> WeatherController? {
+        guard let controller = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "WeatherController") as? WeatherController else { return nil }
+        controller.delegate = self
+        controller.modelWeather = modelCities.arrayWeather[index]
+        return controller
+    }
+
+    func updateArrayCities() {
+        guard let city = self.modelCities.arrayCities.last else { return }
+        let modelWeather = modelCities.getWeatherModel(type: city)
+        modelCities.arrayWeather.append(modelWeather)
+        guard let controller = displayController(index: modelCities.arrayWeather.count - 1) else { return }
+        setViewControllers([controller], direction: .forward, animated: true, completion: nil)
+    }
+    
+    fileprivate func loadDataWeather() {
+        modelCities.updateView = updateArrayCities
+        isEnableLocation() ? modelCities.arrayCities.append(City(name: nil)) : 
```

Comment:
You should not set any default values.
Application should properly process empty `arrayCities` together with disabled location